dotCMS/src/main/java/com/dotcms/rest/api/v1/apps/view/SecretView.java

```diff
@@ -0,0 +1,134 @@
+package com.dotcms.rest.api.v1.apps.view;
+
+import com.dotcms.rest.api.v1.DotObjectMapperProvider;
+import com.dotcms.security.apps.AbstractProperty;
+import com.dotcms.security.apps.ParamDescriptor;
+import com.dotcms.security.apps.Secret;
+import com.dotcms.security.apps.Type;
+import com.fasterxml.jackson.core.JsonGenerator;
+import com.fasterxml.jackson.databind.JsonSerializer;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.SerializerProvider;
+import com.fasterxml.jackson.databind.annotation.JsonSerialize;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+
+@JsonSerialize(using = SecretView.SecretViewSerializer.class)
+public class SecretView {
+
+    private final String name;
+
+    final private Secret secret;
+
+    final private ParamDescriptor paramDescriptor;
+
+    final private boolean dynamic;
+
+    public SecretView(final String name, final Secret secret, final ParamDescriptor paramDescriptor) {
+        this.name = name;
+        this.secret = secret;
+        this.paramDescriptor = paramDescriptor;
+        this.dynamic = null == paramDescriptor;
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public Secret getSecret() {
+        return secret;
+    }
+
+    public ParamDescriptor getParamDescriptor() {
+        return paramDescriptor;
+    }
+
+    public boolean isDynamic() {
+        return dynamic;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+        SecretView that = (SecretView) o;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Local variable 'that' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5329810)